File: test-errors/src/factories/internal/metadata/explore_metadata.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.explore_metadata = void 0;
const Metadata_1 = require("../../../schemas/metadata/Metadata");
const emend_metadata_atomics_1 = require("./emend_metadata_atomics");
const iterate_metadata_1 = require("./iterate_metadata");
const explore_metadata = (checker) => (options) => (collection) => (errors) => (type, explore) => {
    // CONSTRUCT METADATA
    const meta = Metadata_1.Metadata.initialize(explore.escaped);
    if (type === null)
        return meta;
    // ITERATE TYPESCRIPT TYPES
    (0, iterate_metadata_1.iterate_metadata)(checker)(options)(collection)(errors)(meta, type, explore);
    (0, emend_metadata_atomics_1.emend_metadata_atomics)(meta);
    if (meta.escaped) {
        (0, emend_metadata_atomics_1.emend_metadata_atomics)(meta.escaped.original);
        (0, emend_metadata_atomics_1.emend_metadata_atomics)(meta.escaped.returns);
    }
    return meta;
};
exports.explore_metadata = explore_metadata;

File: packages/core/src/factories/internal/metadata/explore_metadata.go
package metadata

func ExploreMetadata(props ExploreMetadataProps) *MetadataSchema {
	metadata := NewMetadataSchema(props.Explore.Escaped)
	if props.Type == nil {
		return metadata
	}
	props.Intersected = false
	IterateMetadata(IMetadataIteratorProps{
		Options: props.Options, Checker: props.Checker, Components: props.Components, Errors: props.Errors,
		Metadata: metadata, Type: props.Type, Explore: props.Explore, Intersected: props.Intersected,
	})
	EmendMetadataAtomics(metadata)
	if metadata.Escaped != nil {
		EmendMetadataAtomics(metadata.Escaped.Original)
		EmendMetadataAtomics(metadata.Escaped.Returns)
	}
	return metadata
}
